lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,298 @@
+		// amount is not increased above MaxFundingAmount.
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount == max-chan-size",
+			amount:             20000000,
+			pushAmt:            lnd.MaxFundingAmount,
+			chanOpenShouldFail: true,
+		},
+
+		// Checks that enough local funds are kept in reserve before
+		// allowing to push amount to remote.
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount == max-chan-size - 1",
+			amount:             20000000,
+			pushAmt:            lnd.MaxFundingAmount - 1,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount 16766000",
+			amount:       20000000,
+			pushAmt:      16766000,
+			carolBalance: lnd.MaxFundingAmount - 16766000,
+		},
+	}
+
+	for _, testCase := range testCases {
+		success := ht.t.Run(
+			testCase.name, func(tt *testing.T) {
+				runTestCase(tt, carol, dave, testCase)
+			},
+		)
+
+		// Stop at the first failure. Mimic behavior of original test
+		// framework.
+		if !success {
+			break
+		}
+	}
+}
+
+func runTestCase(t *testing.T, carol, dave *lntest.HarnessNode,
+	testCase *chanFundMaxTestCase) {
+
+	net.SendCoins(t, testCase.amount, carol)
+	defer func() {
+		if testCase.amount <= 2000 {
+			// Add additional funds to sweep "dust" UTXO.
+			net.SendCoins(ht.t, 100000, carol)
+		}
+
+		// Remove all funds from Carol.
+		sweepNodeWalletAndAssert(carol)
+	}()
+
+	// The parameters to try opening the channel with.
+	chanParams := lntest.OpenChannelParams{
+		Amt:         0,
+		PushAmt:     testCase.pushAmt,
+		SatPerVByte: testCase.feeRate,
+		FundMax:     true,
+	}
+
+	// If we don't expect the channel opening to be
+	// successful, simply check for an error.
+	if testCase.chanOpenShouldFail {
+
+		_, err := net.OpenChannel(
+			carol, dave, chanParams,
+		)
+
+		if err == nil {
+			t.Errorf("should not open channel "+
+				"for: %v", testCase.name)
+		}
+
+		return
+	}
+
+	// Otherwise, if we expect to open a channel use the helper function.
+	chanPoint := openChannelAndAssert(
+		ht, net, carol, dave, chanParams,
+	)
+	defer func() {
+		// Close the channel between Carol and Dave, asserting
+		// that the channel has been properly closed on-chain.
+		closeChannelAndAssert(
+			ht, net, carol, chanPoint, false,
+		)
+	}()
+
+	cType, err := channelCommitType(carol, chanPoint)
+	if err != nil {
+		ht.Fatalf("unable to get channel type: %v", err)
+	}
+
+	// Carol's balance should be her amount subtracted by the
+	// commitment transaciton fee.
+	err = checkChannelBalance(
+		carol, testCase.carolBalance-calcStaticFee(cType, 0),
+	)
+	if err != nil {
+		t.Errorf("carol's balance is wrong: %v", err)
+	}
+
+	// Ensure Dave's balance within the channel is equal to the push amount.
+	err = checkChannelBalance(dave, testCase.pushAmt)
+	if err != nil {
+		t.Errorf("dave's balance is wrong: %v", err)
+	}
+}
+
+// Helper function that returns the fee estimate used for a tx
```

Comment:
nit: not a Godoc comment. And can use the full 80 character width for comments. Same below.